### PR TITLE
Spinnerlidar

### PR DESCRIPTION
**Pull request by alhs6577**
First draft of spinnerlidar model into amr-wind for the RAAW Project

### CODE REVIEW

File: amr-wind/utilities/sampling/DTUSpinnerSampler.cpp
Comment (tonyinme) [inline amr-wind/utilities/sampling/DTUSpinnerSampler.cpp:202]: update comment

**Review by tonyinme — CHANGES_REQUESTED** [review]
Excellent job writing this code @alhs6577 !

Can you please adjust the inputs for this test to make sure it works with this new implementation.
https://github.com/Exawind/amr-wind/blob/main/test/test_files/abl_virtual_lidar/abl_virtual_lidar.i

File: amr-wind/utilities/sampling/DTUSpinnerSampler.cpp
Comment (alhs6577) [reply amr-wind/utilities/sampling/DTUSpinnerSampler.cpp:202]: Whoops, forgot that. The updated test file should be added now.

**Review by tonyinme — APPROVED** [review]

**Comment by tonyinme** [comment]
hey @michaeljbrazell , it looks like this is ready to merge.
@alhs6577 , can you confirm?

**Comment by alhs6577** [comment]
> hey @michaeljbrazell , it looks like this is ready to merge. @alhs6577 , can you confirm?

I'm actually not sure since Nate recently added a couple fast interface function changes to the pull request. Since those haven't been fully tested, there are potentially issues there so we might want to hold off for now.

**Comment by tonyinme** [comment]
> 

oh, yes, I missed the new commits. Thanks for clarifying @alhs6577 !

**Comment by ndevelder** [comment]
Hi @all, This is building and running fine for me on RHEL7 with intel 2019 compiler....but seems to be failing on build tests. Guessing I'm doing something wrong, but having trouble figuring out what. Any thoughts?

**Comment by ndevelder** [comment]
Ok. Added a bunch of idefs for openfast so that all the CI builds would go through. There are a few warnings in Lint-cppcheck that don't have to do with our spinner code. How much more V&V do we want to do before merging this?

**Comment by tonyinme** [comment]
Hi @ndevelder and @alhs6577 , I thought this had been merged a while ago. 
Is this ready to be merged?

**Comment by ndevelder** [comment]
Hi @tonyinme I've been meaning to get back to this, and will try to finalize this afternoon.

**Comment by ndevelder** [comment]
@tonyinme and @alhs6577, this is in the state it was prior to being out of sync with main. I think there could be some more work in the following areas:
- Tests aren't in a great state...but I'm not sure we need to put more work in to merge. 
- Not sure if we need to do anything GPU related...and the cuda test seemingly fails in cmake?
- Documentation and an example case would be nice

**Comment by ndevelder** [comment]
Note, Tommy found that DTU changed the spinner rotational direction, so this PR is on hold until we switch direction and do another verification exercise.

**Comment by ndevelder** [comment]
This work is being continued in https://github.com/Exawind/amr-wind/pull/828

**Pull request closed without merging**
